File: 백준/Silver/9012. 괄호/괄호.py
T = int(input())

for i in range(T):
    stack = []
    isVPS = input()

    for c in isVPS:
        if c == '(':
            stack.append(c)
        elif c == ')':
            if len(stack) == 0:
                stack.append(c)
                break
            else:
                stack.pop()
    
    if len(stack) !=0:
        print("NO")
    else:
        print("YES")

File: 백준/Silver/9012. 괄호/괄호.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n")

const T = Number(input[0]);

for (let i = 1; i <= T; i++) {
  const line = input[i];
  const stack = [];

  for (const c of line){
    if (c == '(') {
      stack.push(c);
    } else if (c == ')') {
      if (stack.length == 0){
        stack.push(c);
        break;
      }
      else{ stack.pop();}
    }
  }

  if (stack.length != 0){
    console.log("NO")
  }else{
    console.log("YES")
  }
}